stubs/evp_md_ctx_is_valid_no_pkey_stub.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <openssl/evp.h>
+
+/*
+ * Checks whether EVP_MD_CTX is a valid object.
+ * Use this stub when we are certain there is no pkey
+ * associated with the digest context.
+ */
+bool evp_md_ctx_is_valid(EVP_MD_CTX *ctx) {
+    assert(ctx->pkey == NULL);
+    return ctx && ctx->is_initialized && ctx->digest_size <= EVP_MAX_MD_SIZE;
+}
```

Comment:
If `ctx` is NULL, there is a null-pointer dereference in line 24.
```c
/*
 * Checks whether EVP_MD_CTX is a valid object.
 */
bool evp_md_ctx_is_valid(EVP_MD_CTX *ctx) {
    if (!ctx) return false;
    /* We must be sure there is no pkey associated with the digest context. */
    assert(ctx->pkey == NULL);
    return ctx->is_initialized && ctx->digest_size <= EVP_MAX_MD_SIZE;
}
```

Reply:
Good catch! Updated.